core/src/main/java/fr/sncf/osrd/envelope_sim/EnvelopePath.java

```diff
@@ -80,7 +80,7 @@ public double findHighGradePosition(double position, double endPos, double lengt
             var grade = getAverageGrade(tailPosition, position);
             if (grade >= gradeThreshold)
                 return position;
-            position += 1.0;
+            position += 10.0;
```

Comment:
Is it a performance thing? (I would guess so)

Reply:
Yes: according to a profiler we spent like 92% of the compute time in this method, and I don't think we need a 1m accuracy to find high slopes. This is just a quick fix, maybe we can find better solutions to improve performances here.

Reply:
yeah we need to do better, thanks for looking into it!